### PR TITLE
V7: Rename autoNotify->autoDetectErrors, add autoDetectUnhandledRejections

### PR DESCRIPTION
**Pull request by bengourley**
Per the updated spec `autoDetectErrors` is now a flag which can be used to disable automatic errors detection of all types.

The additional flag `autoDetectUnhandledRejections` can be used to prevent only promise rejections from being reported, but allow other kinds of error to be reported.

### CODE REVIEW

File: packages/browser/src/notifier.js [hunk 1]
```diff
@@ -62,14 +62,8 @@ module.exports = (opts) => {
   bugsnag.use(pluginSession)
   bugsnag.use(pluginIp)
   bugsnag.use(pluginStripQueryString)
-
-  // optional browser-specific plugins
-
-  if (bugsnag.config.autoNotify !== false) {
```
Comment (bengourley) [inline packages/browser/src/notifier.js]: this logic is now contained within the plugins themselves

**Comment by bengourley** [comment]
The tests on this branch are currently failing for the thing that this commit https://github.com/bugsnag/bugsnag-js/pull/646/commits/cba9d0888efa4ca81092cae4b58a031e455d23df fixes.

I'll rebase this once that lands on `v7` via #646.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## TBD
 
 - Migrate lint tooling to ESLint for both .js and .ts source files [#644](https://github.com/bugsnag/bugsnag-js/pull/644)
+- Rename `autoNotify` -> `autoDetectErrors`, and `autoDetectUnhandledRejections` option [#648](https://github.com/bugsnag/bugsnag-js/pull/648)
```
Comment (djskinner) [inline CHANGELOG.md]: "and" -> "add"?

File: packages/plugin-window-unhandled-rejection/unhandled-rejection.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ const isError = require('@bugsnag/core/lib/iserror')
  */
 let _listener
 exports.init = (client, win = window) => {
+  if (!client.config.autoDetectErrors || !client.config.autoDetectUnhandledRejections) return
```
Comment (djskinner) [inline packages/plugin-window-unhandled-rejection/unhandled-rejection.js]: :+1: I like this pattern over than the conditional loading of plugins

File: packages/plugin-window-onerror/onerror.js [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
 module.exports = {
   init: (client, win = window) => {
+    if (!client.config.autoDetectErrors) return
```
Comment (djskinner) [inline packages/plugin-window-onerror/onerror.js]: Consider adding a test for this

**Pull request merged**